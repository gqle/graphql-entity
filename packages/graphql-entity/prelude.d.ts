export * from './dist/runtime/prelude'
