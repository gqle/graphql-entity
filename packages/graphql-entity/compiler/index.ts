export * from './compile'
export * from './print'
