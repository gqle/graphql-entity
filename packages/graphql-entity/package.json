{
  "name": "@gqle/graphql-entity",
  "version": "0.0.1",
  "author": "Jason Nall <jsonnull@gmail.com>",
  "license": "MIT",
  "bugs": "https://github.com/gqle/graphql-entity/issues",
  "engines": {
    "node": ">=10.0.0"
  },
  "main": "dist/",
  "types": "dist/index.d.ts",
  "files": [
    "/npm-shrinkwrap.json",
    "/dist/*"
  ],
  "scripts": {
    "test": "echo NO TESTS"
  },
  "dependencies": {
    "@gqle/shared": "0.0.1",
    "@graphql-tools/graphql-file-loader": "6.0.12",
    "@graphql-tools/load": "6.0.12",
    "@graphql-tools/schema": "6.1.0",
    "find-up": "4.1.0",
    "graphql": "15.3.0"
  }
}
