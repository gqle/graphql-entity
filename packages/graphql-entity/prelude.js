module.exports = require('./dist/runtime/prelude')
