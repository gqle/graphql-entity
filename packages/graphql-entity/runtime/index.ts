export * from './EntityServer'
export * from './types'
export * from './utils/loadDocuments'
export * from './utils/loadDocumentsByConfig'
export * from './utils/loadSchemaByConfig'
export * from './utils/resolveConfig'
