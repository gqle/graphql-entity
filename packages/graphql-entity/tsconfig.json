{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "composite": true,
    "declaration": true,
    "importHelpers": true,
    "module": "commonjs",
    "outDir": "dist",
    "rootDir": ".",
    "baseUrl": ".",
    "strict": true,
    "target": "es2017"
  },
  "include": ["**/*"],
  "exclude": ["node_modules", "dist"]
}
