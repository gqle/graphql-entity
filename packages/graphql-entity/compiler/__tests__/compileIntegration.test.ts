import { getCompilationFiles } from './getCompilationFiles'

const _defaultSchema = /* GraphQL */ `
  type Query {
    _isExtended: boolean
  }
`

describe('compiler', () => {
  test('basic regression test', async () => {
    const subject = async (source: string) => {
      return getCompilationFiles([['/entity.ts', source]])
    }

    const entitySource = /* GraphQL */ `
      type User {
        name: String!
      }

      extend type Query {
        getUser: User!
      }
    `

    const { files } = await subject(entitySource)

    for (const file of files) {
      expect(file.filepath.path).toMatchSnapshot('file path')
      expect(file.write()).toMatchSnapshot('file contents')
    }
  })

  test('root extensions', async () => {
    const subject = async (source: string) => {
      return getCompilationFiles([['/entity.ts', source]])
    }

    const entitySource = /* GraphQL */ `
      extend type Query {
        getString: String!
      }

      extend type Mutation {
        setString: String!
      }

      extend type Subscription {
        onString: String!
      }
    `

    const { files } = await subject(entitySource)

    for (const file of files) {
      expect(file.filepath.path).toMatchSnapshot('file path')
      expect(file.write()).toMatchSnapshot('file contents')
    }
  })

  // Import of `/entity.ts` from `/` should be `./entity`
  test('relative entity imports', async () => {
    const subject = async (source: string) => {
      return getCompilationFiles([['/entity.ts', source]])
    }

    const entitySource = /* GraphQL */ `
      type User {
        name: String!
      }
    `

    const { files } = await subject(entitySource)

    const rootFile = files.find((f) => f.filepath.path === '/')
    expect(rootFile?.writeImports()).toMatchInlineSnapshot(`
      "import { Awaitable, Maybe, Resolvable } from \\"graphql-entity/prelude\\";
      import { createEntityServer as baseCreateEntityServer } from \\"graphql-entity\\";
      import { User as User } from \\"./entity\\";
      "
    `)
    const entityFile = files.find((f) => f.filepath.path === '/entity.ts')
    expect(entityFile?.writeImports()).toMatchInlineSnapshot(`
      "import { Awaitable, Maybe, Resolvable } from \\"graphql-entity/prelude\\";
      "
    `)
  })
})
