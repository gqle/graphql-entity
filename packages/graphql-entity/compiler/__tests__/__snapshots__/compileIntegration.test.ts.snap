// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiler basic regression test: file contents 1`] = `
"// @gqle/generated
import { Awaitable, Maybe, Resolvable } from \\"graphql-entity/prelude\\";


/*
 * Definitions
 */

export interface User {
  name: Resolvable<string>;
}

/*
 * Entities
 */

export type UserEntity = User


/*
 * RootExtensions
 */

export interface RootExtensions {
  getUser: Resolvable<User>;
}
"
`;

exports[`compiler basic regression test: file contents 2`] = `
"// @gqle/generated
import { Awaitable, Maybe, Resolvable } from \\"graphql-entity/prelude\\";
import { createEntityServer as baseCreateEntityServer } from \\"graphql-entity\\";
import { User as User } from \\"entity\\";


/*
 * Entities
 */

export interface Entities {
  User: User;
}

/*
 * Root
 */

export interface Root {
  getUser: Resolvable<User>;
}

/*
 * Aliases
 */

export type RootEntity = Root

/*
 * Server
 */

export const createEntityServer = (opts: { root: RootEntity }) =>
  baseCreateEntityServer<RootEntity>(opts);
"
`;

exports[`compiler basic regression test: file path 1`] = `"/entity.ts"`;

exports[`compiler basic regression test: file path 2`] = `"/"`;

exports[`compiler root extensions: file contents 1`] = `
"// @gqle/generated
import { Awaitable, Maybe, Resolvable } from \\"graphql-entity/prelude\\";


/*
 * Definitions
 */

/*
 * Entities
 */

/*
 * RootExtensions
 */

export interface RootExtensions {
  getString: Resolvable<string>;
  setString: Resolvable<string>;
  onString: Resolvable<string>;
}
"
`;

exports[`compiler root extensions: file contents 2`] = `
"// @gqle/generated
import { Awaitable, Maybe, Resolvable } from \\"graphql-entity/prelude\\";
import { createEntityServer as baseCreateEntityServer } from \\"graphql-entity\\";


/*
 * Entities
 */

export interface Entities {
}

/*
 * Root
 */

export interface Root {
  getString: Resolvable<string>;
  setString: Resolvable<string>;
  onString: Resolvable<string>;
}

/*
 * Aliases
 */

export type RootEntity = Root

/*
 * Server
 */

export const createEntityServer = (opts: { root: RootEntity }) =>
  baseCreateEntityServer<RootEntity>(opts);
"
`;

exports[`compiler root extensions: file path 1`] = `"/entity.ts"`;

exports[`compiler root extensions: file path 2`] = `"/"`;
